include/netinet/ip.h

```diff
@@ -32,6 +32,33 @@
  * Public Type Definitions
  ****************************************************************************/
 
+#define IPVERSION 4     /* IP version number */
+#define IPDEFTTL  64    /* default ttl, from RFC 1340 */
+
+struct iphdr
+{
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+  unsigned int ihl:4;
+  unsigned int version:4;
+#elif __BYTE_ORDER == __BIG_ENDIAN
+  unsigned int version:4;
+  unsigned int ihl:4;
```

Comment:
I strongly do not recommend using bit fields for network processing

Reply:
glibc and musl both define this bit fields, so many applications use this definition, and if we compile those applications directly in NuttX's environment, we will get a compile error. this structure is only for tripartite library compatibility and is not used by NuttX itself